### PR TITLE
Enable qiskit/tensorcircuit translation with circuit initialization

### PR DESCRIPTION
**Pull request by liwt31**
Enable qiskit/tensorcircuit translation with circuit initialization. The tricky point is that tensorcircuit input may not always be the actual circuit state. So an `enable_inputs` argument is added to `to_qiskit`.

### CODE REVIEW

File: tests/test_circuit.py [hunk 1]
```diff
@@ -1186,6 +1186,34 @@ def cost_fn(_params):
...
+    n = 3
+
+    qis_c = QuantumCircuit(n)
+    qis_c.h(0)
+    qis_c.cnot(0, 1)
+    qis_c.y(2)
+    state = qi.Statevector(qis_c)
+    qis_c = QuantumCircuit(n)
+    qis_c.initialize(state)
+    c = tc.Circuit.from_qiskit(qis_c)
```
Comment (refraction-ray) [inline tests/test_circuit.py:1206]: I suggest add a gate here to make the test more universal (not only initialize instruction) ``c.cx(1, 2)`` and also ``c2.cx(1, 2)`` accordingly

**Comment by refraction-ray** [comment]
>  tensorcircuit input may not always be the actual circuit state

what exactly does this mean?

**Comment by liwt31** [comment]
> > tensorcircuit input may not always be the actual circuit state
> 
> what exactly does this mean?



in `test_qir2qiskit`, the circuit is initialized with
```
n = 6
c = tc.Circuit(n, inputs=tc.array_to_tensor(np.eye(2**n)))
```
which to my knowledge is an input that is not recognizable to Qiskit.

**Comment by refraction-ray** [comment]
> which to my knowledge is an input that is not recognizable to Qiskit.

IC

**Review by refraction-ray — APPROVED** [review]
LGTM

**Pull request merged**